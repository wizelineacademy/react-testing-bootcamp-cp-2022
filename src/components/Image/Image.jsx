import styled from "styled-components";

import formatDate from "../../utils";

const Image = ({ title, date, url }) => (
	<Container>
		<h2>{title}</h2>
		<span>{formatDate(date)}</span>
		<img src={url} alt={title} />
	</Container>
);

const Container = styled.div`
	display: flex;
	flex-direction: column;
	width: 100%;
	h2 {
		text-align: center;
	}
	span {
		text-align: end;
	}
	img {
		width: 100%;
		height: 500px;
	}
`;

export default Image;
