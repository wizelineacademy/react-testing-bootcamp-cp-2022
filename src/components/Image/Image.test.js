import { render, screen } from "@testing-library/react";

import Image from ".";

const setUp = (props) => render(<Image {...props} />);

describe("Image", () => {
	describe("when props are passed", () => {
		it("should display title, date, and image(url)", () => {
			const props = {
				title: "The Full Moon of 2021",
				date: "2022-01-01",
				url: "https://apod.nasa.gov/apod/image/2201/MoonstripsAnnotatedIG_crop1024.jpg",
			};
			setUp(props);

			const imageTitle = screen.getByText(props.title);
			const date = screen.getByText(props.date);
			const image = screen.getByAltText(props.title);

			expect(imageTitle).toBeInTheDocument();
			expect(date).toBeInTheDocument();
			expect(image).toBeInTheDocument();
			expect(image).toHaveAttribute("src", props.url);
		});
	});
});
