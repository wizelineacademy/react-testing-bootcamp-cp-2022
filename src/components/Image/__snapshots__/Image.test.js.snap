// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Image when props are passed should display title, date, and image(url) 1`] = `
<div>
  <div
    class="sc-bczRLJ jgolsG"
  >
    <h2>
      The Full Moon of 2021
    </h2>
    <span>
      01/01/2022
    </span>
    <img
      alt="The Full Moon of 2021"
      height="500"
      src="https://apod.nasa.gov/apod/image/2201/MoonstripsAnnotatedIG_crop1024.jpg"
    />
  </div>
</div>
`;
